frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleAssetPlugin.ts

```diff
@@ -29,62 +29,83 @@ export const validateExtensionsFileSchema = (
+  disableStaticPlugins: pkg.consolePlugin.disableStaticPlugins,
+  extensions: ext,
+});
 
 export class ConsoleAssetPlugin {
-  private readonly manifest: ConsolePluginManifestJSON;
+  private readonly ext: ConsoleExtensionsJSON;
 
-  constructor(private readonly pkg: ConsolePackageJSON) {
-    const ext = parseJSONC<ConsoleExtensionsJSON>(path.resolve(process.cwd(), extensionsFile));
-    validateExtensionsFileSchema(ext).report();
-
-    this.manifest = {
-      name: pkg.consolePlugin.name,
-      version: pkg.consolePlugin.version,
-      displayName: pkg.consolePlugin.displayName,
-      description: pkg.consolePlugin.description,
-      dependencies: pkg.consolePlugin.dependencies,
-      disableStaticPlugins: pkg.consolePlugin.disableStaticPlugins,
-      extensions: ext,
-    };
+  constructor(
+    private readonly pkg: ConsolePackageJSON,
+    private readonly remoteEntryCallback: string,
+    private readonly skipExtensionValidator = false,
+  ) {
+    this.ext = parseJSONC<ConsoleExtensionsJSON>(path.resolve(process.cwd(), extensionsFile));
+    validateExtensionsFileSchema(this.ext).report();
   }
 
   apply(compiler: webpack.Compiler) {
-    compiler.hooks.shouldEmit.tap(ConsoleAssetPlugin.name, (compilation) => {
-      const result = new ExtensionValidator(extensionsFile).validate(
-        compilation,
-        this.manifest.extensions,
-        this.pkg.consolePlugin.exposedModules || {},
+    compiler.hooks.thisCompilation.tap(ConsoleAssetPlugin.name, (compilation) => {
+      // Generate additional assets
+      compilation.hooks.processAssets.tap(
+        {
+          name: ConsoleAssetPlugin.name,
+          stage: webpack.Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL,
+        },
+        () => {
+          compilation.emitAsset(
+            pluginManifestFile,
+            new webpack.sources.RawSource(
+              Buffer.from(JSON.stringify(getPluginManifest(this.pkg, this.ext), null, 2)),
+            ),
+          );
+        },
       );
 
-      if (result.hasErrors()) {
-        result.getErrors().forEach((e) => {
-          // TODO(vojtech): revisit after bumping webpack 5 to latest stable version
-          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-          // @ts-ignore
-          compilation.errors.push(new Error(e));
-        });
-        return false;
-      }
-      return true;
+      // Post-process assets already present in the compilation
+      compilation.hooks.processAssets.tap(
+        {
+          name: ConsoleAssetPlugin.name,
+          stage: webpack.Compilation.PROCESS_ASSETS_STAGE_ADDITIONS,
+        },
+        () => {
+          compilation.updateAsset(remoteEntryFile, (source) => {
+            const newSource = new webpack.sources.ReplaceSource(source);
+            newSource.insert(
+              this.remoteEntryCallback.length + 1,
+              `'${this.pkg.consolePlugin.name}@${this.pkg.consolePlugin.version}', `,
+            );
```

Comment:
Not new, but see #9198.

Reply:
OK, we can first find the position of `remoteEntryCallback` and insert from there.

Reply:
PR updated to address this issue.